### PR TITLE
Create component Cart,MenuItem,OrderConfirmation

### PR DESCRIPTION
**Pull request by Zeicks**

### CODE REVIEW

File: src/main/java/com/easyrest/components/menuPage/Cart.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class Cart {
+
+    private final By submitOrder = By.xpath("//*[text()='Submit order']");
+    private final By showCartButton = By.xpath("//button [@aria-label = 'Show cart']");
+    private final By deleteSecondItemButton = By.xpath("//div[contains(@class , 'MuiPaper-root')][2]//div[contains(@style ,'' )]//button [@aria-label ='Remove item']");
+
+    protected WebDriver driver;
```
Comment (ViktoriaMilkevich) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: Driver should be private, not protected

File: src/main/java/com/easyrest/components/menuPage/Cart.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public Cart submitOrder() {
+        driver.findElement(submitOrder).click();
+        return this;
+    }
+
+    public Cart clickOnButtonShowCart() {
+        driver.findElement(showCartButton).click();
+        return this;
```
Comment (ViktoriaMilkevich) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: Does your DOM tree change after clicking?

File: src/main/java/com/easyrest/components/menuPage/MenuItem.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+public class MenuItem {
+
+    private final By addProduct = By.xpath("//button[@aria-label = 'Add to cart']");
+
+
+    protected WebDriver driver;
```
Comment (ViktoriaMilkevich) [inline src/main/java/com/easyrest/components/menuPage/MenuItem.java]: Driver should be private

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class OrderConfirmation {
+
+    private final By submitInWindowButton = By.xpath("//*[text()='Submit']");
+    private final By cancelInWindowButton = By.xpath("//span[text()='Cancel']");
+    private final By deleteDishInOrderButton = By.xpath("//*[contains(@class, 'MuiTableRow-root')][1]//button[@aria-label= 'Remove item']");
+
+    protected WebDriver driver;
```
Comment (ViktoriaMilkevich) [inline src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java]: should be private

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public OrderConfirmation cancelOrderInWindow() {
+        driver.findElement(cancelInWindowButton).click();
+        return new OrderConfirmation(driver);
+    }
+
+    public OrderConfirmation clickOnButtonDeleteFirstDish() {
+        driver.findElement(deleteDishInOrderButton).click();
+        return new OrderConfirmation(driver);
+    }
```
Comment (ViktoriaMilkevich) [inline src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java:30]: Components should be created in the MenuPage and not exist by their own

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 1 above)
Comment (martadiachok) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: decided to change on private

Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: ok

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package com.easyrest.components.menuPage;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class OrderConfirmation {
```
Comment (martadiachok) [inline src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java:6]: I think in this class should be locator to field time picker and date picker and also methods that click on time picker and return component of time picker and  second method that click on date picker field and return the component of calendar

Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java:6]: Yes, but at this stage I cannot add them, and we do not need them for the test case.

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public OrderConfirmation submitOrderInWindow() {
+        driver.findElement(submitInWindowButton).click();
+        return new OrderConfirmation(driver);
+    }
+
+    public OrderConfirmation cancelOrderInWindow() {
+        driver.findElement(cancelInWindowButton).click();
+        return new OrderConfirmation(driver);
```
Comment (vadymtsehelny) [inline src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java]: After cancel the OrderConfirmation window disappears, so maybe the return should not be the OrderConfirmation?

File: src/main/java/com/easyrest/components/menuPage/Cart.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public Cart clickOnButtonDeleteSecondItem() {
```
Comment (dmatsev) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: why do you delete always the second item? is the second item always present on the page?

File: src/main/java/com/easyrest/components/menuPage/Cart.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    private final By showCartButton = By.xpath("//button [@aria-label = 'Show cart']");
+    private final By deleteSecondItemButton = By.xpath("//div[contains(@class , 'MuiPaper-root')][2]//div[contains(@style ,'' )]//button [@aria-label ='Remove item']");
+
+    public Cart(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public Cart submitOrder() {
+        driver.findElement(submitOrder).click();
+        return this;
```
Comment (dmatsev) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: do we stay on the same page after submitting an order?

File: src/main/java/com/easyrest/components/menuPage/Cart.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public Cart clickOnButtonShowCart() {
```
Comment (dmatsev) [inline src/main/java/com/easyrest/components/menuPage/Cart.java]: Does `showCartButton` exist on the Cart page?

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    private final By cancelInWindowButton = By.xpath("//span[text()='Cancel']");
+    private final By deleteDishInOrderButton = By.xpath("//*[contains(@class, 'MuiTableRow-root')][1]//button[@aria-label= 'Remove item']");
+
+    public OrderConfirmation(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public OrderConfirmation submitOrderInWindow() {
+        driver.findElement(submitInWindowButton).click();
+        return new OrderConfirmation(driver);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java]: Do we stay in the confirmation window after submitting an order?

File: src/main/java/com/easyrest/components/menuPage/MenuItem.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    private final By addProduct = By.xpath("//button[@aria-label = 'Add to cart']");
+
+    public MenuItem(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public MenuItem addProduct() {
+        List<WebElement> elements = driver.findElements(addProduct);
+        if (elements.size() > 0) {
+            WebElement addition = elements.get(0);
```
Comment (dmatsev) [inline src/main/java/com/easyrest/components/menuPage/MenuItem.java:21]: What does `addition` mean?

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 9 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: yes, we stay on the same page but over this page we get component OrderConfirmation

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 10 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: Fixed

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 8 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: I removed it as we don't need it

File: src/main/java/com/easyrest/components/menuPage/MenuItem.java (hunk 12 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/MenuItem.java:21]: addition - element that is added to the order

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 10 above)
Comment (dmatsev) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: What is an answer?

File: src/main/java/com/easyrest/components/menuPage/MenuItem.java (hunk 12 above)
Comment (dmatsev) [reply src/main/java/com/easyrest/components/menuPage/MenuItem.java:21]: What is an answer?

File: src/main/java/com/easyrest/pages/MenuPage.java [hunk 13]
```diff
@@ -1,140 +1,7 @@
...
 
-import com.sun.jna.WString;
-import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.WebElement;
-
-import java.util.List;
 
 public class MenuPage extends BasePage {
-
```
Comment (dmatsev) [inline src/main/java/com/easyrest/pages/MenuPage.java:11]: why did you delete a lot of code?

**Review by dmatsev — CHANGES_REQUESTED** [review]
Resolve the comments.

File: src/main/java/com/easyrest/pages/MenuPage.java (hunk 13 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/pages/MenuPage.java:11]: > why did you delete a lot of code?

because we don't need so much code

**Review by ViktoriaMilkevich — APPROVED** [review]

**Review by martadiachok — APPROVED** [review]

**Review by vadymtsehelny — APPROVED** [review]

File: src/main/java/com/easyrest/components/menuPage/Cart.java (hunk 2 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/Cart.java]: Yes, DOM tree changes. Fixed

File: src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java (hunk 5 above)
Comment (Zeicks) [reply src/main/java/com/easyrest/components/menuPage/OrderConfirmation.java:30]: ok

**Comment by martadiachok** [comment]
Incorrect file in pull-request, rewrite this to another https://github.com/martadiachok/EasyRest-Testing/pull/28

File: src/main/java/com/easyrest/pages/MenuPage.java [hunk 14]
```diff
@@ -1,140 +1,19 @@
...
-    private final By hotCategoryMenuItem = By.xpath("//span [text() ='Hot']");
-    private final By pizzaCategoryMenuItem = By.xpath("//span [text() ='Pizza']");
-    private final By meatCategoryMenuItem = By.xpath("//span [text() ='Meat']");
-    private final By pastaCategoryMenuItem = By.xpath("//span [text() ='Pasta']");
-    private final By bakeryCategoryMenuItem = By.xpath("//span [text() ='Bakery']");
-    private final By hardLiquerCategoryMenuItem = By.xpath("//span [text() ='Hard liquer']");
-    private final By deleteSecondItemButton = By.xpath("//div[contains(@class , 'MuiPaper-root')][2]//div[contains(@style ,'' )]//button [@aria-label ='Remove item']");
+    private Cart cart;
+    private MenuItem menuItem;
+    private OrderConfirmation orderConfirmation;
```
Comment (vadymtsehelny) [inline src/main/java/com/easyrest/pages/MenuPage.java:12]: Just check if you can use these private instance variables, maybe you need some methods
